pygsp/graphs/fourier.py

```diff
@@ -88,26 +103,43 @@ def compute_fourier_basis(self, recompute=False):
         True
 
         """
+        if n_eigenvectors is None:
+            n_eigenvectors = self.N
 
-        if hasattr(self, '_e') and hasattr(self, '_U') and not recompute:
+        if hasattr(self, '_e') and hasattr(self, '_U') and not recompute \
+                and n_eigenvectors <= len(self.e):
             return
 
         assert self.L.shape == (self.N, self.N)
-        if self.N > 3000:
+        if self.N > 3000 and n_eigenvectors == self.N:
```

Comment:
Shall we change this test to something like: self.N*n_eigenvectors > 3000**2 ?

Reply:
Agreed. It should ideally represent the true computational cost. I know it's O(N**3) for the full eigendecomposition. Do you know for the partial? I.e. dependency on matrix size and number of eigenvectors.

Reply:
I've had a look and estimating the complexity of `scipy.sparse.linalg.eigsh` (which uses the Lanczos method from ARPACK) is not trivial. My best guess is to use N^2 k, though this is not based on any solid knowledge.

Reply:
Ok. I don't know either. I agree on N^2 k as a rough estimate.